File: src/01-easy/mime-type/solution.lua
local n = tonumber(io.read())
local q = tonumber(io.read())
local m = {}
for i = 1, n do
    local line = io.read()
    local ext, mt = line:match("(%S+)%s+(%S+)")
    m[ext:lower()] = mt
end
for i = 1, q do
    local f = io.read()
    local ext = f:match("%.([^.]+)$")
    if ext then
        local r = m[ext:lower()]
        print(r or "UNKNOWN")
    else
        print("UNKNOWN")
    end
end

File: src/01-easy/mime-type/solution.rs
use std::collections::HashMap;
use std::io::{self, BufRead, Write, BufWriter};
fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut lines = stdin.lock().lines();
    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let q: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let mut m = HashMap::new();
    for _ in 0..n {
        let line = lines.next().unwrap().unwrap();
        let mut parts = line.split_whitespace();
        let ext = parts.next().unwrap().to_lowercase();
        let mt = parts.next().unwrap().to_string();
        m.insert(ext, mt);
    }
    for _ in 0..q {
        let f = lines.next().unwrap().unwrap();
        let f = f.trim_end();
        match f.rfind('.') {
            None => writeln!(out, "UNKNOWN").unwrap(),
            Some(dot) => {
                let ext = f[dot+1..].to_lowercase();
                writeln!(out, "{}", m.get(&ext).map(|s| s.as_str()).unwrap_or("UNKNOWN")).unwrap();
            }
        }
    }
}
